model/src/cfd.rs

```diff
@@ -1403,10 +1379,45 @@ impl Cfd {
         }))
     }
 
+    fn event_with_reason(&self, event: EventKind, reason: anyhow::Error) -> CfdEvent {
+        self.log_cfd_event(&event, Some(reason));
+        self.issue_event_internal(event)
+    }
+
     fn event(&self, event: EventKind) -> CfdEvent {
+        self.log_cfd_event(&event, None);
+        self.issue_event_internal(event)
+    }
+
+    fn issue_event_internal(&self, event: EventKind) -> CfdEvent {
         CfdEvent::new(self.id, event)
     }
 
+    fn log_cfd_event(&self, event: &EventKind, reason: Option<anyhow::Error>) {
```

Comment:
especially here `reason` could be renamed to something more substantial, e.g. `detail` or `context`